packages/adapter-drizzle/src/lib/pg.ts

```diff
@@ -239,18 +227,295 @@ export function PostgresDrizzleAdapter(
   }
 }
 
-export type PostgresTableFn<T extends TableConfig> = PgTableWithColumns<{
-  name: T["name"]
-  columns: T["columns"]
-  dialect: T["dialect"]
+export type DefaultPostgresUsersTable = PgTableWithColumns<{
+  name: string
+  columns: {
+    id: PgColumn<{
+      name: string
+      columnType: "PgVarchar" | "PgText"
```

Comment:
If I understood correctly, this is restricting the column type to these two types.

What if the user wants to use an `autoincrement` id for example?

Reply:
Honestly, I am not sure if it is should be allowed to specify `non-string` type.


@ndom91  Can you please specify what type the `id` might be in the `users` table? Unfortunately, the information about the models at this [link](https://authjs.dev/getting-started/adapters#models) is currently unavailable, but as far as I remember, the `id` should be a `string`. In the TypeScript interface `AdapterUser`, the `id` is also a `string`.

Reply:
probably safest to only allow a single typescript datatype. could be a potential maintenance hell if the core lib should start allowing different data types if they reference user.id and can't expect it to always be of a certain datatype.

on the typesafety side, if you'd accept number here, you'd also want to restrict that `user.id` has the same type as `account.userId` etc etc and that can quickly escalate in complexity

Reply:
There's a more general page regarding DB adapters now available here: https://authjs.dev/concepts/database-models

I agree that `user.id: String` makes most sense :+1:

Reply:
Also more useful info here as well: https://authjs.dev/guides/creating-a-database-adapter

Reply:
You are right, this change was introduced on next-auth v4.
Before that we could use number there.

Sorry for the confusion.